### PR TITLE
techdebt: [PIE-2607]: Added JOOQ Queries for Pipeline Overview Dashboard

### PR DESCRIPTION
**Pull request by shalini-agr**
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30436)

### CODE REVIEW

**Review by vaibhav-si — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30436)* status: 1 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PipelineDashboardServiceImpl.java, line 63 at r1](https://reviewable.io/reviews/harness/harness-core/30436#-MutWYsZ1Cfnvmeajuyb:-MutWYsZ1Cfnvmeajuyc:b-o457cv) ([raw file](https://github.com/harness/harness-core/blob/5fa7164d279e7685ccff6d6f3e95f5d0f334c6da/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PipelineDashboardServiceImpl.java#L63)):*
> ```Java
>     public String queryBuilderSelectStatusAndTime(String accountId, String orgId, String projectId, String pipelineId,
>                                                   long startInterval, long endInterval, String tableName) {
>         Query selectStatusQuery = this.dsl.select(Tables.PIPELINE_EXECUTION_SUMMARY_CD.STATUS, Tables.PIPELINE_EXECUTION_SUMMARY_CD.STARTTS).where(Tables.PIPELINE_EXECUTION_SUMMARY_CD.ACCOUNTID.eq(accountId).and(Tables.PIPELINE_EXECUTION_SUMMARY_CD.ORGIDENTIFIER.eq(orgId)).and(Tables.PIPELINE_EXECUTION_SUMMARY_CD.PIPELINEIDENTIFIER.eq(pipelineId)).and(Tables.PIPELINE_EXECUTION_SUMMARY_CD.PROJECTIDENTIFIER.eq(projectId)).and(Tables.PIPELINE_EXECUTION_SUMMARY_CD.STARTTS.ge(startInterval)).and(Tables.PIPELINE_EXECUTION_SUMMARY_CD.STARTTS.lt(endInterval)).and(accountId != null).and(orgId != null).and(projectId != null).and(pipelineId != null).and(startInterval > 0).and(endInterval > 0));
> ```

run clang formatter
___
*[800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PipelineDashboardServiceImpl.java, line 72 at r1](https://reviewable.io/reviews/harness/harness-core/30436#-MutWSjw4UWj6mEVwPoM:-MutWSjw4UWj6mEVwPoN:b-xq7dr4) ([raw file](https://github.com/harness/harness-core/blob/5fa7164d279e7685ccff6d6f3e95f5d0f334c6da/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PipelineDashboardServiceImpl.java#L72)):*
> ```Java
> //    }
> //
> //    if (orgId != null) {
> ```

delete these lines, no need to comment. Previous version is anyways available in Git

File: 800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/QueryService.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+import org.jooq.DSLContext;
+import org.jooq.impl.DSL;
+
+public class QueryService {
+  @Inject private DSLContext dsl;
+  @Inject TimeScaleDBService timeScaleDBService;
+
+  private String CD_TableName = "pipeline_execution_summary_cd";
+
+  public StatusAndTime queryBuilderSelectStatusAndTime(String accountId, String orgId, String projectId,
```
Comment (vaibhav-si) [inline 800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/QueryService.java:29]: can we rename these methods? We are no longer builder the query.

Make it something like getPipelineExecutionStatusAndTime

Comment (vaibhav-si) [reply 800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/QueryService.java:29]: Similarly rename other methods

**Comment by shalini-agr** [comment]
trigger messageCheck

**Comment by shalini-agr** [comment]
trigger messagecheck

**Comment by shalini-agr** [comment]
trigger authorcheck

**Review by vaibhav-si — CHANGES_REQUESTED** [review]
> Reviewed 1 of 6 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30436)* status: 16 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/QueryService.java, line 45 at r6](https://reviewable.io/reviews/harness/harness-core/30436#-MvHlAQZDstPwH8PIUwZ:-MvHlAQZDstPwH8PIUw_:bgv2nzq) ([raw file](https://github.com/harness/harness-core/blob/02376480e23b83d36c74c734a1e42d47d1d90da0/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/QueryService.java#L45)):*
> ```Java
>                          .and(Tables.PIPELINE_EXECUTION_SUMMARY_CD.STARTTS.ge(startInterval))
>                          .and(Tables.PIPELINE_EXECUTION_SUMMARY_CD.STARTTS.lt(endInterval))
>                          .and(accountId != null)
> ```

all below null checks can be removed, also >0 check can be removed
___
*[800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/QueryService.java, line 52 at r6](https://reviewable.io/reviews/harness/harness-core/30436#-MvHlyiP51dRopOf5nf2:-MvHlyiP51dRopOf5nf3:bud1m71) ([raw file](https://github.com/harness/harness-core/blob/02376480e23b83d36c74c734a1e42d47d1d90da0/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/QueryService.java#L52)):*
> ```Java
>                          .and(endInterval > 0))
>               .fetch()
>               .into(PipelineExecutionSummaryCd.class);
> ```

directly fetch into statusAndTime

**Review by vaibhav-si — COMMENTED** [review]
> Reviewed 7 of 9 files at r2.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30436)* status: 16 file(s) to review. 1 change requests. 0 of 2 approvals obtained.

**Review by shalini-agr — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30436)* status: 16 file(s) to review. 1 change requests. 5. unresolved discussions. 0 of 2 approvals obtained. 

___
*[800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PipelineDashboardServiceImpl.java, line 63 at r1](https://reviewable.io/reviews/harness/harness-core/30436#-MutWYsZ1Cfnvmeajuyb:-MvJ8YZCASK4DFaArbAV:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/5fa7164d279e7685ccff6d6f3e95f5d0f334c6da/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PipelineDashboardServiceImpl.java#L63)):*
<details><summary><i>Previously, vaibhav-si (Vaibhav Singhal) wrote…</i></summary><blockquote>

run clang formatter
</blockquote></details>

Done.
___
*[800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PipelineDashboardServiceImpl.java, line 72 at r1](https://reviewable.io/reviews/harness/harness-core/30436#-MutWSjw4UWj6mEVwPoM:-MvJ8ZO3BUNt0LlVAiNt:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/5fa7164d279e7685ccff6d6f3e95f5d0f334c6da/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PipelineDashboardServiceImpl.java#L72)):*
<details><summary><i>Previously, vaibhav-si (Vaibhav Singhal) wrote…</i></summary><blockquote>

delete these lines, no need to comment. Previous version is anyways available in Git
</blockquote></details>

Done.
___
*[800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/QueryService.java, line 29 at r6](https://reviewable.io/reviews/harness/harness-core/30436#-MvHfJHxQeE2n-ga2nrc-r6-29:-MvJ8We0ExCKqN20DTJo:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/02376480e23b83d36c74c734a1e42d47d1d90da0/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/QueryService.java#L29)):*
<details><summary><i>Previously, vaibhav-si (Vaibhav Singhal) wrote…</i></summary><blockquote>

Similarly rename other methods
</blockquote></details>

Done.
___
*[800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/QueryService.java, line 45 at r6](https://reviewable.io/reviews/harness/harness-core/30436#-MvHlAQZDstPwH8PIUwZ:-MvJ8MSbDAyKK6IuZ8Er:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/02376480e23b83d36c74c734a1e42d47d1d90da0/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/QueryService.java#L45)):*
<details><summary><i>Previously, vaibhav-si (Vaibhav Singhal) wrote…</i></summary><blockquote>

all below null checks can be removed, also >0 check can be removed
</blockquote></details>

Done.
___
*[800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/QueryService.java, line 52 at r6](https://reviewable.io/reviews/harness/harness-core/30436#-MvHlyiP51dRopOf5nf2:-MvJ8VeQ7oXmGDmEnoKa:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/02376480e23b83d36c74c734a1e42d47d1d90da0/800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/QueryService.java#L52)):*
<details><summary><i>Previously, vaibhav-si (Vaibhav Singhal) wrote…</i></summary><blockquote>

directly fetch into statusAndTime
</blockquote></details>

Done.

**Review by prashantsharma04 — APPROVED** [review]
> Reviewed 7 of 9 files at r2, 1 of 3 files at r3, 1 of 3 files at r5, 2 of 6 files at r7, 2 of 2 files at r9, 3 of 3 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30436)* status: 8 file(s) to review. 1 change requests. 5. unresolved discussions. 1 of 2 approvals obtained.

**Review by vaibhav-si — APPROVED** [review]
> Reviewed 1 of 3 files at r5, 1 of 6 files at r7, 2 of 2 files at r9, 3 of 3 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30436)* status: 1 file(s) to review.

**Review by vaibhav-si — APPROVED** [review]
> Reviewed 1 of 3 files at r3.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30436)* status: :shipit: complete! ✓

**Pull request merged**